sphinx/source/docs/dev_guide/bokehjs.rst

```diff
@@ -121,98 +121,226 @@ Use ``node make help`` to list all available commands.
 You can use ``tsc`` directly for error checking (e.g. in an IDE). However, don't use
 it for code emit, because we rely on AST transforms to produce viable library code.
 
-Testing
-~~~~~~~
+.. _devguide_bokehjs_development_testing:
 
-BokehJS testing is performed with the ``node make test`` command. You can run individual
-test suites with ``node make test:suite_name``. Known tests suites are:
+Test suites
+~~~~~~~~~~~
 
-* ``node make test:codebase``
-* ``node make test:defaults``
-* ``node make test:unit``
-* ``node make test:integration``
+BokehJS comes with its own suites of tests and testing framework. All tests for BokehJS
+use ``describe()`` and ``it()`` functions. Unit tests are written using Chai "expect"
+style. [TBD: How are style/setup different for visual integration tests?]
 
-The last two can be run with ``node make test:lib``. Unit and integration tests are
-run in a web browser (see requirements), which is started automatically with the
-right settings to guarantee consistent test results.
+To launch BokehJS tests, run ``node make test`` from within the
+:bokeh-tree:`bokehjs/test` directory.
 
-To review the visual tests' output, start BokehJS's devtools server:
+Instead of running all available BokehJS tests, you can also run individual test
+suites with ``node make test:suite_name``. Available tests suites are:
+
+* ``node make test:codebase``: Codebase tests checking file size limits
+* ``node make test:defaults``: Tests checking whether the defaults in Bokeh's
+  Python models match those of Bokeh's TypeScript models
+* ``node make test:unit``: Unit tests for BokehJS
+* ``node make test:integration``:
+  :ref:`Visual integration tests <devguide_bokehjs_development_visual_testing>`
+  comparing locally generated plots against a set of baseline files
+
+You can combine the last two test suites by running ``node make test:lib``.
+
+Unit and integration tests are run in a web browser (see requirements). The test suite
+automatically starts the web browser with the right settings to ensure consistent test
+results.
+
+.. _devguide_bokehjs_development_devtoolsserver:
+
+Testing with Devtool Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to tests run from the command line, you can also use the BokehJS's devtools
+server. Use this system to run tests and review the visual tests' output.
+
+First, start the devtools server with the following command:
 
 .. code-block:: sh
 
     $ node test/devtools server
     listening on 127.0.0.1:5777
 
-and navigate to ``/integration/report``. Devtools server can also be used to
-manually inspect and debug tests. For that, the following endpoints are available:
+Next, open the displayed server URL in a webbrowser and navigate to
+``/integration/report``, ``/unit/report``, or ``/defaults/report``.
```

Comment:
@mattpap Does the devtools server work with all three URLs? I was only able to get a report with ``/integration/report`` but I believe that is due to some issues with my local node environment...

Reply:
I only found the `/integration/report` route defined inside `bokehjs/test/devtools/server.ts`

Reply:
I agree, I cannot get `/unit/report` or `/defaults/report`.  I am on Ubuntu 20.10 with a pretty standard conda/node setup.

Reply:
Currently report is generated only when `--baseline-root` option is used with bokehjs' devtools, which is enabled only for integration tests. I suppose I wanted to generate reports in all cases, but didn't get to it.

Reply:
Thanks, I'll remove ``/unit/report`` and ``/defaults/report``.